File: vet/src/pojo/ProdutoBiologico.java
package pojo;

/**
 * POJO 4 — ProdutoBiologico
 * Subclasse de ProdutoVeterinario.
 * Representa produtos de origem biológica (vacinas, soros, antígenos).
 */
public class ProdutoBiologico extends ProdutoVeterinario {

    private String tipoAgente;     // Ex: "Vírus atenuado", "Bactéria inativada"
    private String sorotipo;       // Ex: "H1N1", "Sorotipo B"
    private int    numDoses;       // Número de doses no frasco

    public ProdutoBiologico() {}

    public ProdutoBiologico(int id, String nome, double preco, String fabricante,
                            String registroMapa, String especieAlvo, String viaAdministracao,
                            String tipoAgente, String sorotipo, int numDoses) {
        super(id, nome, preco, fabricante, registroMapa, especieAlvo, viaAdministracao);
        this.tipoAgente = tipoAgente;
        this.sorotipo   = sorotipo;
        this.numDoses   = numDoses;
    }

    // ── Getters / Setters ────────────────────────────────────────────────────

    public String getTipoAgente()                      { return tipoAgente; }
    public void   setTipoAgente(String tipoAgente)     { this.tipoAgente = tipoAgente; }

    public String getSorotipo()                        { return sorotipo; }
    public void   setSorotipo(String sorotipo)         { this.sorotipo = sorotipo; }

    public int    getNumDoses()                        { return numDoses; }
    public void   setNumDoses(int numDoses)            { this.numDoses = numDoses; }

    @Override
    public String toString() {
        return "ProdutoBiologico{" +
               "id=" + getId() +
               ", nome='" + getNome() + '\'' +
               ", tipoAgente='" + tipoAgente + '\'' +
               ", sorotipo='" + sorotipo + '\'' +
               ", numDoses=" + numDoses + '}';
    }
}

File: cliente-cpp_entrega_1/src/ProdutoBiologico.cpp
#include "ProdutoBiologico.hpp"

ProdutoBiologico::ProdutoBiologico(){};
ProdutoBiologico::ProdutoBiologico(int id, const string& nome, double preco, const string& fabricante,
    const string& regM, const string& espA, const string& vAdm,    
    const string& tipoAgente, const string& sorotipo, int numDoses) 
    : ProdutoVeterinario(id, nome, preco, fabricante, regM, espA, vAdm),
    tipoAgente(tipoAgente), sorotipo(sorotipo), numDoses(numDoses) {}

const string& ProdutoBiologico::getTipoAgente() const{
    return tipoAgente;
}

void ProdutoBiologico::setTipoAgente(const string& tipoAgente){
    this->tipoAgente = tipoAgente;
}

const string& ProdutoBiologico::getSorotipo() const{
    return sorotipo;
}

void ProdutoBiologico::setSorotipo(const string& sorotipo){
    this->sorotipo = sorotipo;
}

int ProdutoBiologico::getNumDoses() const{
    return numDoses;
}

void ProdutoBiologico::setNumDoses(int numDoses){
    this->numDoses = numDoses;
}

string ProdutoBiologico::toString() const{
    return "ProdutoBiologico{id=" + to_string(this->getId()) + 
    ", nome='" + this->getNome() +
    "', tipoAgente='" + tipoAgente + 
    "', sorotipo='" + sorotipo + 
    "', numDoses=" + to_string(numDoses) + "}";
}
